src/Microsoft.AspNetCore.Mvc.Core/Problem.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A machine-readable format for specifying errors in HTTP API responses based on https://tools.ietf.org/html/rfc7807.
+    /// </summary>
+    public class Problem
+    {
+        /// <summary>
+        /// A URI reference [RFC3986] that identifies the problem type. This specification encourages that, when
+        /// dereferenced, it provide human-readable documentation for the problem type
+        /// (e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be
+        /// "about:blank".
+        /// </summary>
+        public string Type { get; set; }
+
+        /// <summary>
+        /// A short, human-readable summary of the problem type.It SHOULD NOT change from occurrence to occurrence
+        /// of the problem, except for purposes of localization(e.g., using proactive content negotiation; 
+        /// see[RFC7231], Section 3.4).
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// The HTTP status code([RFC7231], Section 6) generated by the origin server for this occurrence of the problem. 
+        /// </summary>
+        public int? Status { get; set; }
+
+        /// <summary>
+        /// A human-readable explanation specific to this occurrence of the problem. 
+        /// </summary>
+        public string Detail { get; set; }
+
+        /// <summary>
+        /// A URI reference that identifies the specific occurrence of the problem.It may or may not yield further information if dereferenced.
+        /// </summary>
+        public string Instance { get; set; }
+
+        public object this[string key]
+        {
+            get
+            {
+                AdditionalProperties.TryGetValue(key, out var value);
+                return value;
+            }
+            set => AdditionalProperties[key] = value;
+        }
+
+        public IDictionary<string, object> AdditionalProperties { get; } = new Dictionary<string, object>(StringComparer.Ordinal);
```

Comment:
Should this be lazily-initialized with a property indicating it's been created? Not sure always creating the dictionary is much better than subclassing `SerializableError` or another dictionary type.